cotton-usb-host/src/wire.rs

```diff
@@ -346,27 +346,35 @@ pub fn parse_descriptors(buf: &[u8], v: &mut impl DescriptorVisitor) {
 
         match dtype {
             CONFIGURATION_DESCRIPTOR => {
+                dlen = Ord::min(dlen, size_of::<ConfigurationDescriptor>());
```

Comment:
Have you found a real device which requires this? If so, would it be possible to dump its config descriptors so that we can write a unit-test? if you can plug the device into a Linux box, you could get the descriptors a bit like this:
```
 $ sudo python3
 > import usb
 > dev = usb.core.find(idVendor=0x424, idProduct=0x4064) # as needed
 > usb.control.get_descriptor(dev, 0x400, usb.util.DESC_TYPE_CONFIG, 0)
```

Reply:
Added test case with CFG from Yamaha KBD that has oversize EP descriptors. Removed checks from CFG and IF since no test case.

Reply:
Cool, thanks! In fact I just checked the spec, and hosts are supposed to silently truncate any oversize descriptors, so I'll fix up that version. But in any case it feels right to name the device that's getting it wrong!